0.2-Elliptic-Curve-Math.ipynb

```diff
@@ -0,0 +1,303 @@
+    "b_key.set(b.to_bytes(32,'big'), True) \n",
+    "ab_key = a_key * b_key\n",
+    "\n",
+    "# Ensure operations are equivalent.\n",
+    "print(int().from_bytes(ab_key.get_bytes(),'big') == ab)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 1.3 _Programming Exercise:_ Commutative property of scalar operations\n",
+    "\n",
+    "In this exercise we wish to demonstrate the commutative property of scalar addition and multiplication, whilst getting familiarized with both integer modulo operations and the private key `ECKey` methods.\n",
+    "\n",
+    "Consider: \n",
+    "* `a + b = b + a` over SECP256k1\n",
+    "* `a * b = b * a` over SECP256k1\n",
+    "\n",
+    "Notes:\n",
+    "* Demonstrate that both equations hold.\n",
+    "* Compute the left sides with the Python `int` class.\n",
+    "* Compute the right sides with the Bitcoin Core `ECKey` class.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = random.randrange(1, SECP256K1_ORDER)\n",
+    "b = random.randrange(1, SECP256K1_ORDER)\n",
+    "\n",
+    "# Left: Compute a + b\n",
+    "\n",
+    "# Right: Compute b + a\n",
+    "\n",
+    "# Left/Right: Assert equality\n",
+    "\n",
+    "\n",
+    "# Left: Compute a * b\n",
+    "\n",
+    "# Right: Compute b * a\n",
+    "\n",
+    "# Left/Right: Assert equality\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 1.4 _Programming Exercise:_ Distributivity of scalar operations\n",
+    "\n",
+    "In this exercise we wish to demonstrate the distributivity property of scalar addition and multiplication.\n",
+    "\n",
+    "Consider: `(a - b) * c = a * c - b * c` over SECP256k1\n",
+    "* Demonstrate that the equation holds\n",
+    "* Compute the left side with the Python `int` class.\n",
+    "* Compute the right side with the Bitcoin Core `ECKey` class.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = random.randrange(1, SECP256K1_ORDER)\n",
+    "b = random.randrange(1, SECP256K1_ORDER)\n",
+    "c = random.randrange(1, SECP256K1_ORDER)\n",
+    "\n",
+    "# Left: Compute (a - b) * c\n",
+    "\n",
+    "\n",
+    "# Right: Compute -b * c\n",
+    "# Negate a ECKey with negate() method.\n",
+    "\n",
+    "\n",
+    "# Right: Compute a * c\n",
+    "\n",
+    "\n",
+    "# Right: Compute a * c - b * c\n",
+    "\n",
+    "\n",
+    "# Left/Right: Assert equality\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 1.5 Example: Point Addition over SECP256k1\n",
+    "\n",
+    "The public key `ECPubkey` class can be derived from `ECKey` with the `ECKey.get_pubkey` method. \n",
+    "\n",
+    "In the following example, we perform point addition with `ECPubkey.__add__`.\n"
```

Comment:
You don't need to mention `__add__()` here. Just say that 'we perform point addition'.

Reply:
fixed